### PR TITLE
Implement test stat

### PR DESCRIPTION
**Pull request by orvend**
Statistical hypothesis tests implementation

### CODE REVIEW

File: probdist/binomial.go [hunk 1]
```diff
@@ -1,28 +1,39 @@
...
 	P float64
 }
 
-// Mean returns the mean of the binomial distribution
+// NewBinomial is used to initialize binomial parameters. What is different from
+// Binomial type is that here the parameters are validated.
+// N must be an integer positive number and P a real between 0 and 1.
+func NewBinomial(n float64, p float64) (Binomial, error) {
+	if math.Mod(n, 1.0) != 0 || p < 0 || p > 1.0 {
+		return Binomial{20.0, 0.5}, errors.New("stats: invalid Binomial parameters")
```
Comment (qmuntal) [inline probdist/binomial.go]: Why do you initialize the struct? Just return Binomial{}.

Same applies to other constructors.

File: probdist/normal.go [hunk 2]
```diff
@@ -1,43 +1,142 @@
...
-// Variance returns the Variance of the normal distribution
+// Variance returns the Variance of the normal distribution.
 func (norm Normal) Variance() float64 {
 	return math.Pow(norm.Sigma, 2.0)
 }
+
+//Data from: https://www.itl.nist.gov/div898/handbook/eda/section3/eda3671.htm.
+var zStandardNormal statsTable = statsTable{
+	[][]float64{
+		[]float64{0.00000, 0.00399, 0.00798, 0.01197, 0.01595, 0.01994, 0.02392, 0.02790, 0.03188, 0.03586},
```
Comment (qmuntal) [inline probdist/normal.go]: I think that it is not necessary to repeat `[]float64` in every member of the array.

If this table has a fixed number of rows and columns, better use an array of array than a slice of slices, so the compiler can optimize it and even store it in the stack.

File: probdist/normal.go [hunk 3]
```diff
@@ -1,43 +1,142 @@
...
+		[]float64{0.49966, 0.49968, 0.49969, 0.49970, 0.49971, 0.49972, 0.49973, 0.49974, 0.49975, 0.49976},
+		[]float64{0.49977, 0.49978, 0.49978, 0.49979, 0.49980, 0.49981, 0.49981, 0.49982, 0.49983, 0.49983},
+		[]float64{0.49984, 0.49985, 0.49985, 0.49986, 0.49986, 0.49987, 0.49987, 0.49988, 0.49988, 0.49989},
+		[]float64{0.49989, 0.49990, 0.49990, 0.49990, 0.49991, 0.49991, 0.49992, 0.49992, 0.49992, 0.49992},
+		[]float64{0.49993, 0.49993, 0.49993, 0.49994, 0.49994, 0.49994, 0.49994, 0.49995, 0.49995, 0.49995},
+		[]float64{0.49995, 0.49995, 0.49996, 0.49996, 0.49996, 0.49996, 0.49996, 0.49996, 0.49997, 0.49997},
+		[]float64{0.49997, 0.49997, 0.49997, 0.49997, 0.49997, 0.49997, 0.49998, 0.49998, 0.49998, 0.49998},
+	},
+
+	[]float64{
```
Comment (qmuntal) [inline probdist/normal.go]: If this slice fixed number of elements, better use an array than a slice, so the compiler can optimize it and even store it in the stack.

File: probdist/normal.go [hunk 4]
```diff
@@ -1,43 +1,142 @@
...
+		3.4,
+		3.5,
+		3.6,
+		3.7,
+		3.8,
+		3.9,
+		4.0,
+	},
+
+	[]float64{0.00, 0.01, 0.02, 0.03, 0.04, 0.05, 0.06, 0.07, 0.08, 0.09},
```
Comment (qmuntal) [inline probdist/normal.go]: If this row has a fixed number of elements, better use an array than a slice, so the compiler can optimize it and even store it in the stack.

File: probdist/statstables.go [hunk 5]
```diff
@@ -0,0 +1,37 @@
+package stats
+
+type statsTable struct {
+	table [][]float64
```
Comment (qmuntal) [inline probdist/statstables.go]: Use an array instead of a slice

File: probdist/statstables.go [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+	if c0 > tab.col[0] || c0 < tab.col[len(tab.col)-1] {
+		panic("stats: c0 value not valid in the table")
+	}
+
+	for i, a := range tab.col[:len(tab.col)-1] {
+		if c0 <= a {
+			l0 = a
+			r0 = float64(tab.col[i+1])
+			lid = i
+			rid = i + 1
```
Comment (qmuntal) [inline probdist/statstables.go]: I think it is more readable this approach:
```go
l0, r0 = a, float64(tab.col[i+1])
lid, rid = i, i+1
```

Comment (orvend) [reply probdist/statstables.go]: I agree

**Pull request merged**